development/sourcemap-validator.js

```diff
@@ -86,16 +86,27 @@ async function validateSourcemapForFile ({ buildName }) {
       const portion = line.slice(result.column)
       const isMaybeValid = portion.includes(targetString)
       if (!isMaybeValid) {
-        console.error('Sourcemap seems invalid:')
-        console.log(`\n========================== ${result.source} ====================================\n`)
-        console.log(line)
-        console.log(`\n==============================================================================\n`)
+        console.error(`Sourcemap seems invalid:\n${getFencedCode(result.source, line)}`)
       }
     })
   })
   console.log(`  checked ${sampleCount} samples`)
 }
 
+const CODE_FENCE_LENGTH = 80
+const TITLE_PADDING_LENGTH = 1
+
+function getFencedCode (filename, code) {
+  const title = `${' '.repeat(TITLE_PADDING_LENGTH)}${filename}${' '.repeat(TITLE_PADDING_LENGTH)}`
+  const openingFenceLength = Math.max(CODE_FENCE_LENGTH - (filename.length + (TITLE_PADDING_LENGTH * 2)), 0)
+  const startOpeningFenceLength = Math.floor(openingFenceLength / 2)
+  const endOpeningFenceLength = Math.ceil(openingFenceLength / 2)
+  const openingFence = `${'='.repeat(startOpeningFenceLength)}${title}${'='.repeat(endOpeningFenceLength)}`
+  const closingFence = '='.repeat(CODE_FENCE_LENGTH)
+
+  return `${openingFence}\n${code}\n${closingFence}\n`
+}
+
```

Comment:
I see the vision, but wouldn't it be better to just hardcode the `===...` strings inside the function?

Reply:
I'm not sure what you mean; better how? Hard-coding the `===...` doesn't allow making the opening and closing fence to be the same length, for one thing.

Reply:
Simpler, but I'm not gonna block I'm on it 😛

Reply:
My original edit was something like:

```
        console.error(`
Sourcemap seems invalid:
========================== ${result.source} ====================================
${line}
==============================================================================\n`)
```
But there wasn't any simple way of dealing with the awkward indentation, and the mismatched lengths bothered me. I figured this string-builder approach would adequately handle both problems in a way that was still fairly readable, despite being more complex? :man_shrugging: